src/libs/polycube/src/table.cpp

```diff
@@ -36,13 +37,23 @@ class RawTable::impl {
   int first(void *key);
   int next(const void *key, void *next);
 
+  int pop(void *value);
+  int push(const void *value);
+
  private:
+  bool is_queue_stack_enabled;
+  bool is_batch_enabled;
   int fd_;
 };
 
-RawTable::impl::~impl() {}
+RawTable::impl::~impl() {
+}
 
-RawTable::impl::impl(void *op) : fd_(*(int *)op) {}
+RawTable::impl::impl(void *op) : fd_(*(int *)op) {
+  auto version = get_kernel_release();
+  is_queue_stack_enabled = compare_kernel_release(version, QUEUE_STACK_KERNEL_RELEASE);
```

Comment:
Honestly, I would prefer the other way, i.e., create a different map for the Queue and Stack.
I have a question, what is the behavior of the `bpf_map_lookup` with those tables? And, on the other hand, what is the behavior of the `bpf_map_lookup_and_delete` with the other tables?

Reply:
Perfect, I will change the implementation according to your suggestion :)
According to https://github.com/torvalds/linux/blob/24085f70a6e1b0cb647ec92623284641d8270637/kernel/bpf/syscall.c#L1442 :
* `bpf_map_lookup_and_delete` returns ENOSUPP if the map is not Queue/Stack
* `bpf_map_lookup` seems not to work with Queue/stack (table.next() returns always -1)

Reply:
Solved in 523ff8b 
Let me know if this new solution is appreciated @sebymiano